get_http.go

```diff
@@ -223,6 +266,63 @@ func (g *HttpGetter) parseMeta(r io.Reader) (string, error) {
 	}
 }
 
+// getByteRange is a helper function to parse out the byte range for ranged
+// requests.
+//
+// input values:
+// u must be non-nil and unmodified.
+// example of u:
+// "http://my/file.iso?ranged_request_bytes=5555-66666"
+// "http://my/file.iso?ranged_request_bytes=5555-"
+// note that the format of the raw url string can be either
+// bytes=startByte- OR bytes=startByte-endByte
+//
+// return values:
+// (non-nil, non-nil) - should not occur
+// (nil, nil) - returned when no byte range was provided in the first place
+// (nil, non-nil) - error occured when parsing or validating the byte range
+// (non-nil, nil) - byte range was successfully parsed, and the range is
+// returned as an array of strings in the following format:
+// example non-nil return val "vals":
```

Comment:
🤔 I'm not sure what line 286 means?

Reply:
lol the name of the value we're returning is "vals" but you're right that reads like absolute gibberish.